viewing_party/party.py

```diff
@@ -1,23 +1,146 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if title and genre and rating:
+        movie_dict = {'title': title, 'genre': genre, 'rating': rating}
+    else:
+        return None
+    
+    return movie_dict
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_dict, movie_title):
+    movie_watched = None
+    for i in range(len(user_dict["watchlist"])):
+        if user_dict["watchlist"][i]["title"] == movie_title:
+            movie_watched = user_dict["watchlist"].pop(i)
+    if movie_watched == None:
+        return user_dict
+    else:
+        user_dict["watched"].append(movie_watched)
+    return user_dict
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_dict):
+    rating_total = 0
+    if user_dict["watched"]:
+        for i in range(len(user_dict["watched"])):
+            rating_total += user_dict["watched"][i]["rating"]
+            avg_rating = rating_total / (i + 1)

```

Comment:
Try to avoid using indexing when all that's needed is the value. Indexes are often a source of errors. Consider
```python
        for movie in user_dict["watched"]:
            rating_total += movie["rating"]

        avg_rating = rating_total / len(user_dict["watched"])
```

Also notice that we don't need to calculate the average every time through the loop.